### DISCUSSION TITLE
Better book reading support

### DISCUSSION THREAD

**Discussion by yegle** [iSoron/uhabits#1288] [Feature Requests]
Hi,

I recently start using this app to track the progress of reading books to train a habit out of it. The experience is not great and I think I have a couple ideas to improve it.

## Book reading progress is a Counter but the app treats it as Gauge

When adding a measurable habit, the question can only be framed to ask for a gauge values (how many pages did you read today), whereas the progress of reading of a book is inherently a Counter (I'm on page 145 of 354).

It would be awesome if the question can be framed as "what's the last page you read" then calculate the delta since the last answer, and provide option to either 1) spread the # of books read over the days since last answer, or 2) assume the pages were read on the day of answering the question.

This improvement can also benefit other use cases. E.g. if we want to develop a habit of driving a car, we can frame the question as "what's the current odometer reading as of now?".

## Reading multiple books and contribute to the same habit

Once the previous FR is addressed, the natural next step would be reading multiple books concurrently and contribute "pages read" to the same habit.

E.g. I can have habit "read book A" and "read book B" both associate with habit "read more books", and the number of pages read in the sub-habits will automatically contribute to the "read more books" habit. An important thing is to make the sub-habit disposable/ephemeral, i.e. I do not need the statistics to show of the 1000 pages read, how many are read on book A and how many are on book B.

To reuse the silly example above, one can drive multiple cars to learn the habit of driving cars, we can have one sub-habit per car and track them separately.

Perhaps a more realistic example would be tracking a "Running" habit, while a non-zero answer to "how many miles did you run today" automatically answers "did you exercise today" question for a "more exercises" habit.

**Comment by iSoron** [comment]
Thanks for the suggestions, @yegle. I agree that it's currently not very easy to track how many pages you read; you need to memorize where did you stop, then do some mental math. Perhaps the app could help somehow, but it's a bit hard for me to see how could that be done in the user interface. Adding individual books would be too troublesome, and the app is not really optimize for ephemeral habits, as you mention. I am open to suggestions. Just to mention a workaround, starting in Loop 2.1, you will be able to add notes to entries, so you should be able to write down what book did you read, and what page did you stop at.

**Reply by yegle** [reply]
That's a very useful workaround. Will the note be included in the reminder or in the UI to type in the numbers?

**Reply by isgeles** [reply]
I think it would be cool if there is a "Total" value for the targets like there is one for "Today", "Week",...
That total value could be used to track the pages you read and if you start a new book, you can create a new habit.
It would also fit well into the UI.

**Reply by yegle** [reply]
Hmm, is the 2.1 version out yet? I'm still running 2.0.3 installed from Play Store.

**Comment by navinthomas** [comment]
@yegle There is an app called [link to non-open source app removed] which might be a more suitable option for your use case. Unfortunately, the free tier allows you to add only 10 books. But there is a paid tier which removes this limitation, but it can be quite expensive. Despite that, I'd recommend you to give it a try,
